stdlib/result.mc

```diff
@@ -348,6 +348,40 @@ utest
   ()
 with ()
 
+let _foldl : all w. all e. all a. all b.
+  (a -> b -> (Result w e a)) -> a -> [b] -> Result w e a
+  = lam f. lam acc.
+    recursive
+      let workOK : Map Symbol w -> a -> [b] -> Result w e a
+        = lam accWarn. lam acc. lam seq.
+          match seq with [hd] ++ tl then
+            switch f acc hd
+            case ResultOk c then
+              workOK (mapUnion accWarn c.warnings) (c.value) tl
+            case ResultErr c then
+              ResultErr {c with warnings = mapUnion accWarn c.warnings }
+            end
+          else
+            ResultOk { warnings = accWarn, value = acc }
+    in workOK (_emptyMap ()) acc
+
+let _foldlfun : all w. all e. all a. all b.
```

Comment:
This does not feel general enough to have in the library here I think. I think it would be implemented with

```
lam a. lam b. lam fs. result.foldlM (lam a. lam f. f a b) a fs
```

looking purely at the type signature

Reply:
Agreed, moved it to `composition-check.env` with your definition.